### PR TITLE
GRF-789 : Delete attribute endpoint + command + command handler

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+use Akeneo\Category\Domain\Query\DeleteTemplateAttribute;
+use Akeneo\Category\Domain\ValueObject\Attribute\AttributeUuid;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateUuid;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class DeleteCategoryTemplateAttributeSql implements DeleteTemplateAttribute
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php:17]: Am I correct in assuming that this service is not used anywhere yet?

File: src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public function __construct(
+        private readonly ValidatorInterface $validator,
+        private readonly DeactivateAttribute $deactivateAttribute,
+    ) {
+    }
+
+    public function __invoke(DeleteAttributeCommand $command): void
+    {
+        $violations = $this->validator->validate($command);
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php:27]: What validations are we doing here for an attribute deletion?

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php (hunk 1 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php:17]: Yes that's true, I implemented it by mistake and I let it this way to be used by the [GRF-791].(https://akeneo.atlassian.net/browse/GRF-791)

File: src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php (hunk 2 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php:27]: There is none for now, I decided on the moment to let it to follow the same kind of implementation than the AddAttributeCommandeHandler.

But I can delete this useless part.

Comment (etiennekerbastard) [reply src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php:27]: Yes. Either come up with validations specific to attribute deletion (if you're stumped, let's discuss it during a daily tech, or ask @antoinetrouve as he knows symfony validations well) or remove it completely. A good rule of thumb is to not implement dead code that's not used anywhere :)

Comment (amigaire) [reply src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php:27]: I would suggest to validate that the attribute uuid we try to delete is effectively linked to the given template uuid

Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php:27]: Good idea @amigaire :) !

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php (hunk 1 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php]: I finally pushed push it in the GRF-791 directly.

File: src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php (hunk 2 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Application/Command/DeleteAttributeCommandHandler.php]: After discussion, we decided that this was not necessary to validate anything with Synfony validators for that case (which is quite simple).

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public function delete(TemplateUuid $templateUuid, AttributeUuid $attributeUuid): void
+    {
+        if (($this->isTemplateDeactivated)($templateUuid)) {
+            return;
+        }
+
+        $query = <<< SQL
+            DELETE FROM pim_catalog_category_attribute
+            WHERE category_template_uuid = UUID_TO_BIN(:template_uuid) 
+            AND uuid = UUID_TO_BIN(:attribute_uuid);
```
[the remark this answers is not in the document]
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/DeleteCategoryTemplateAttributeSql.php]: We decided to keep both arguments for clarity to make the debugging and the understanding easier for others.

**Review by etiennekerbastard — APPROVED** [review]

**Review by amigaire — APPROVED** [review]

**Pull request merged**